pkg/driver/node_server.go

```diff
@@ -1964,6 +1965,35 @@ func (driver *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoReque
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	// Decode and check if the node is configured
+	nodeInfo, err := driver.flavor.GetNodeInfo(nodeID)
+	if err != nil {
+		log.Error("Cannot unmarshal node from node ID. err: ", err.Error())
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
+
+	nodeLabels, err := driver.flavor.GetNodeLabelsByName(nodeInfo.Name)
+	log.Infof("node %s nodeLabels: %+v", nodeInfo.Name, nodeLabels)
+	if err != nil {
+		log.Error("Cannot load node labels from node. err: ", err.Error())
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
+
+	accessableTopologySegments := make(map[string]string)
+	for key, value := range nodeLabels {
+		log.Infof("node %s label: %s => %s", nodeInfo.Name, key, value)
+		if key == "csi.hpe.com/zone" {
```

Comment:
Would you mind moving the key string into `constants.go` and name it something along the lines of `defaultCSITopologyKey` as I can imagine we want to parameterize this in the future and allow customers to use existing topology keys.